File: output/ChatGPT/avatar/Java/C++/atcoder_ABC123_C.cpp
#include <iostream>
#include <queue>
#include <cmath>
using namespace std;

int main() {
    long long N;
    cin >> N;
    long long min = 0;
    for (int i = 0; i < 5; i++) {
        if (i == 0)
            cin >> min;
        else
            cin >> min;
            min = min(min, min);
    }
    cout << ceil((double)N / min) + 4 << endl;

}

File: output/ChatGPT/avatar/Python/Java/atcoder_ABC123_C.java
import java.util.Scanner;

public class atcoder_ABC123_C {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int[] A = new int[5];
        
        for (int i = 0; i < 5; i++) {
            A[i] = scanner.nextInt();
        }
        
        int result = (int) Math.ceil((double) N / min(A)) + 4;
        System.out.println(result);
    }
    
    public static int min(int[] arr) {
        int minVal = arr[0];
        
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] < minVal) {
                minVal = arr[i];
            }
        }
        
        return minVal;
    }
}